nix/pkgs/server.nix

```
@@ -0,0 +1,57 @@
+{
+  lib,
+  clangStdenv,
+  boost,
+  clang,
+  cmake,
+  openssl,
+  libmysqlclient,
+  readline,
+  bzip2,
+  git,
+  modules ? [ ],
+}:
+
+clangStdenv.mkDerivation {
+  name = "azerothcore-wotlk";
+
+  src = ../../.;
+
+  nativeBuildInputs = [ cmake ];
+
+  # The name of the directory needs to have the same name as the module it's used when linking
+  patchPhase = ''
+    for i in ${lib.concatStringsSep " " modules};
+    do
+        for j in $(ls -1 $i);
+        do
+          ln -s $i/$j modules/
```

I'm not strongly familiar with nix (though I use guix) but I typically prefer shell code to wrap variables in quotes

```suggestion
    for i in ${lib.concatStringsSep " " modules};
    do
        for j in "$(ls -1 "$i")";
        do
          ln -s "$i/$j" modules/
```

---

I didn't quote things here because the module names cannot have spaces in their names as they are using as their symbol names in the code.

A space in the folder name would end up breaking compilation later on.

---

ah, that makes sense. That's fine.